Automattic/wp-calypso
# Posts: Leverage QueryPosts lifecycle methods to ensure posts are always requested

This pull request seeks to offer an alternative solution to the problem described at https://github.com/Automattic/wp-calypso/pull/2350#issuecomment-172629151 . Rather than dispatching an action to reset query caches upon navigating to the editor, the changes here leverage lifecycle methods of React components to ensure that requests are always triggered upon mount and upon receiving different props. As such, when a user returns to the editor screen, regardless of whether query results have been cached during a previous visit, a network request will always be made to retrieve the current set of posts. Because a cached result is known, we can present these options until the updated set has been received.

__Implementation notes:__

fb27f21 and 1301280 are simply revert commits of the solution offered at #2350. Code review can be focused on 5ca4033 and 80e8d06.

__Testing instructions:__

Verify that there are no regressions in the behavior described for the parent page selector in the testing instructions at #2350. Specifically, ensure that after creating a page, if you navigate away from the editor and later return to draft a new page, the page that you had created previously is offered as an option for parent page.

/cc @mtias
